engine/crates/composition/src/compose/entity_interface.rs

```diff
@@ -188,3 +164,28 @@ pub(crate) fn merge_entity_interface_definitions<'a>(
         }
     }
 }
+
+fn translate_arguments(
+    field: subgraphs::Walker<'_, (subgraphs::FieldId, subgraphs::FieldTuple)>,
+    ctx: &mut ComposeContext<'_>,
+) -> federated::InputValueDefinitions {
+    let mut ids: Option<federated::InputValueDefinitions> = None;
+    for arg in field.arguments() {
+        let directives = collect_composed_directives(std::iter::once(arg.directives()), ctx);
+        let name = ctx.insert_string(arg.name().id);
+        let id = ctx.insert_input_value_definition(ir::InputValueDefinitionIr {
+            name,
+            r#type: arg.r#type().id,
+            directives,
+            description: None,
+        });
+
+        if let Some((_start, len)) = &mut ids {
+            *len += 1;
+        } else {
+            ids = Some((id, 1));
+        }
+    }
```

Comment:
I find these loops a bit awkward. Worth adding an abstraction?

Reply:
What I do for these kind of thinds in engine-v2 is:
```
let start = array.len();
// insert stuff
let end = array.len();
(start, end - start)
```

you could also have a `insert_input_value_definitions()` taking an `Iterator` for argument